src/ngraph_backend_config.cc

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "ngraph_backend_config.h"
+
+using namespace std;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+BackendConfig::BackendConfig(string backend_name) {
+  NGRAPH_VLOG(3) << "BackendConfig() ";
+  backend_name_ = backend_name;
+  optional_attributes_ = {"_ngraph_device_config"};
+}
+
+string BackendConfig::join(unordered_map<string, string> optional_parameters) {
+  NGRAPH_VLOG(3) << "JOIN";
+  // If _ngraph_device_config is not found
+  // throw an error
```

Comment:
We do not have an assert on the optional_parameters size, as it can be greater than 1. For eg in NNPI we have 4 attributes being passed though only one is being used to create the string.

We are throwing an error if the attribute we are interested in is not passed.

Reply:
Since this is an optional attribute - why are we then throwing exception when the attribute not present?

Reply:
This is more like additional attributes that are backend specific. I can change the name.